frontend/src/employee-frontend/components/messages/Sidebar.tsx

```diff
@@ -64,6 +64,13 @@ function Accounts(props: AccountsParams) {
   const [selectedUnit, setSelectedUnit] = useState<SelectOptionProps>(
     unitOptions[0]
   )
+  useEffect(() => {
+    props.view &&
+      props.setView({
+        ...props.view,
+        selectedUnitId: props.view.account.daycareGroup?.unitId
+      })
+  }, [props.view?.account.daycareGroup?.unitId])
```

Comment:
What is happening here? 🤔 If we get a view that has an account, immediately set the view again only converting the view to include unit? Shouldn't this happen elsewhere?

Moreover, props will change when any prop change so this will trigger on all renders.